src/main/webapp/js/planner.js

```diff
@@ -26,14 +26,46 @@ async function getPlan() {
     selectedClasses: selectedClasses,
     semesters: document.getElementById('semesters').value
   };
-  const response = await fetch('/api/planner', {
-    method: 'POST',
-    body: JSON.stringify(data),
-  });
+  courseContainer.innerText = '';
+  let response;
+  let courseList;
+  try {
+    response = await fetch('/api/planner', {
+      method: 'POST',
+      body: JSON.stringify(data),
+    });
+    courseList = await response.json();
+  } catch (err) {
+    alert('An error occurred.');
+    return;
+  }
+  if (response.ok) {
```

Comment:
isn't this implicitly ok because no error was thrown?

Reply:
the errors are only thrown if there's an error when getting the content, so this catches any responses with a status that is not in the 200s.

Reply:
ah ok, sounds good, thanks for the clarification.